lightning/src/routing/scoring.rs

```diff
@@ -707,12 +717,21 @@ impl<G: Deref<Target = NetworkGraph<L>>, L: Deref, T: Time> Score for Probabilis
 			return u64::max_value();
 		}
 
-		if let EffectiveCapacity::ExactLiquidity { liquidity_msat } = usage.effective_capacity {
-			if usage.amount_msat > liquidity_msat {
-				return u64::max_value();
-			} else {
-				return self.params.base_penalty_msat;
-			};
+		let mut anti_probing_penalty_msat = 0;
+		match usage.effective_capacity {
+			EffectiveCapacity::ExactLiquidity { liquidity_msat } => {
+				if usage.amount_msat > liquidity_msat {
+					return u64::max_value();
+				} else {
+					return self.params.base_penalty_msat;
+				}
+			},
+			EffectiveCapacity::Total { capacity_msat, htlc_maximum_msat: Some(htlc_maximum_msat) } => {
+				if htlc_maximum_msat*2 >= capacity_msat {
```

Comment:
> Hmm, looking at this again do we need to handle the *2 overflow? I assume not because it's bounded by the channel capacity, which is itself bounded by 21 million? 
Right, that was my line of thought. I assume we'd never come close to `u64::MAX`?
That said, I'll just go the division route to make sure.

Reply:
Done in https://github.com/lightningdevkit/rust-lightning/pull/1555/commits/800ccec0ed1e925a6918efacb966d7a91d484308.